File: API/RemoteServiceServer.cs
using System;
using System.Web;
using AustinHarris.JsonRpc;
using ACRemote;

namespace ACRemote.API {
	public class RemoteServiceServer : JsonRpcService, IRemoteService {
		IRemote remote = null;
		IDHT11 temp = null;
		public RemoteServiceServer (IRemote rmt, IDHT11 temp) {
			this.remote = rmt;
			this.temp = temp;
		}
		[JsonRpcMethod]
		public string[] __interface__() {
			System.Collections.ArrayList methods = new System.Collections.ArrayList();
			foreach (string p in new string[]{"set", "get"}) {
				foreach (string m in new string[]{"power", "temp", "mode", "speed"}) {
					methods.Add(string.Format("%s_%s", p, m));
				}
			}
			methods.Add("actual_temp");
			methods.Add("actual_humidity");
			return (string[])methods.ToArray();
		}
		[JsonRpcMethod]
		public void set_power(bool val) {
			remote.power = (bool)val;
		}
		[JsonRpcMethod]
		public bool get_power() {
			return remote.power;
		}
		[JsonRpcMethod]
		public void set_temp(int val) {
			remote.temp = val;
		}
		[JsonRpcMethod]
		public int get_temp() {
			return remote.temp;
		}
		[JsonRpcMethod]
		public String get_mode() {
			return remote.mode.ToString();
		}
		[JsonRpcMethod]
		public void set_mode(string val) {
			remote.mode = (ACRemote.modes)System.Enum.Parse(typeof(ACRemote.modes), val);
		}
		[JsonRpcMethod]
		public String get_speed() {
			return remote.speed.ToString();
		}
		[JsonRpcMethod]
		public void set_speed(string val) {
			remote.speed = (ACRemote.speeds)System.Enum.Parse(typeof(ACRemote.speeds), (String)val);
		}
		[JsonRpcMethod]
		public double actual_temp() {
			return temp.temp;
		}
		[JsonRpcMethod]
		public double actual_humidity() {
			return temp.humidity;
		}
	}
}

File: API/RemoteServiceServer.vb
' JSON-RPC service server
' Author: Kevin Ross

Imports AustinHarris.JsonRpc
Imports ACRemote.ACRemote

Namespace ACRemote.API
    Public Class RemoteServiceServer
        Inherits JsonRpcService
        Implements IRemoteService
        Private remote As IRemote = Nothing
        Private temp As IDHT11 = Nothing
        Public Sub New(rmt As IRemote, temp As IDHT11)
            Me.remote = rmt
            Me.temp = temp
        End Sub
        <JsonRpcMethod>
        Public Sub set_power(val As Boolean) Implements IRemoteService.set_power
            remote.power = val
        End Sub
        <JsonRpcMethod>
        Public Function get_power() As Boolean Implements IRemoteService.get_power
            Return remote.power
        End Function
        <JsonRpcMethod>
        Public Sub set_temp(val As Integer) Implements IRemoteService.set_temp
            remote.temp = val
        End Sub
        <JsonRpcMethod>
        Public Function get_temp() As Integer Implements IRemoteService.get_temp
            Return remote.temp
        End Function
        <JsonRpcMethod>
        Public Function get_mode() As String Implements IRemoteService.get_mode
            Return remote.mode.ToString()
        End Function
        <JsonRpcMethod>
        Public Sub set_mode(val As String) Implements IRemoteService.set_mode
            remote.mode = DirectCast(System.Enum.Parse(GetType(modes), val), modes)
        End Sub
        <JsonRpcMethod>
        Public Function get_speed() As String Implements IRemoteService.get_speed
            Return remote.speed.ToString()
        End Function
        <JsonRpcMethod>
        Public Sub set_speed(val As String) Implements IRemoteService.set_speed
            remote.speed = DirectCast(System.Enum.Parse(GetType(speeds), DirectCast(val, [String])), speeds)
        End Sub
        <JsonRpcMethod>
        Public Function actual_temp() As Double Implements IRemoteService.actual_temp
            Return temp.temperature
        End Function
        <JsonRpcMethod>
        Public Function actual_humidity() As Double Implements IRemoteService.actual_humidity
            Return temp.humidity
        End Function
        <JsonRpcMethod>
        Public Sub __set_temp(val As Double) Implements IRemoteService.__set_temp
        End Sub
        <JsonRpcMethod>
        Public Sub __set_humidity(val As Double) Implements IRemoteService.__set_humidity
        End Sub
        <JsonRpcMethod>
        Public Sub reset() Implements IRemoteService.reset
            remote.reset()
        End Sub
    End Class
End Namespace

